packages/starknet-snap/src/ui/components/ExecuteTxnUI.tsx

```diff
@@ -0,0 +1,207 @@
+    // Therefore, we use DEFAULT_DECIMAL_PLACES set to 18 here. If additional fee tokens with different decimals are introduced,
+    // this logic should be updated to handle token-specific decimals dynamically.
+    tokenTotals[selectedFeeToken] = {
+      amount: feeTokenAmount,
+      decimals: DEFAULT_DECIMAL_PLACES,
+    };
+  }
+
+  return tokenTotals;
+};
+
+/**
+ * A component for executing transactions, providing details and options to configure the transaction.
+ *
+ * @param props - The component props.
+ * @param props.signer - The signer for the transaction.
+ * @param props.chainId - The ID of the chain for the transaction.
+ * @param props.maxFee - The maximum fee allowed for the transaction.
+ * @param props.calls - The calls involved in the transaction.
+ * @param props.selectedFeeToken - The token used for fees.
+ * @param props.includeDeploy - Whether to include account deployment in the transaction.
+ * //param props.errors : TODO : add this param
+ * @returns The ExecuteTxnUI component.
+ */
+export const ExecuteTxnUI: SnapComponent<ExecuteTxnUIProps> = ({
+  signer,
+  chainId,
+  maxFee,
+  calls,
+  selectedFeeToken,
+  includeDeploy,
+  // errors, // TODO: include this later
+}) => {
+  // Calculate the totals using the helper
+  const tokenTotals = calculateTotals(calls, maxFee, selectedFeeToken);
+
+  return (
+    <Container>
+      <Box>
+        <Section>
+          <AddressUI label="Signer" address={signer} chainId={chainId} />
+        </Section>
+
+        {/* Loop through each call and render based on isTransfer */}
+
+        {calls.map((call) => (
+          <Section>
+            {call.isTransfer ? (
+              <AddressUI
+                label="Token Transfer"
+                address={call.contractAddress}
+                chainId={chainId}
+              />
+            ) : (
+              <AddressUI
+                label="Contract Interaction"
+                address={call.contractAddress}
+                chainId={chainId}
+              />
+            )}
+            {call.transferAmount &&
+            call.transferSenderAddress &&
+            call.transferRecipientAddress &&
+            call.isTransfer ? (
+              <Section>
+                {call.transferSenderAddress === signer ? null : (
+                  <AddressUI
+                    label="Sender Address"
+                    address={call.transferSenderAddress}
+                    chainId={chainId}
+                  />
+                )}
+                <AddressUI
+                  label="Recipient Address"
+                  address={call.transferRecipientAddress}
+                  chainId={chainId}
+                />
+                <Row label={`Amount`}>
+                  <Text>{`${formatUnits(
+                    call.transferAmount,
+                    call.transferTokenDecimals,
+                  )} ${call.transferTokenSymbol as string}`}</Text>
+                </Row>
```

Comment:
```suggestion
                <Amount label={`Amount`} amount={call.transferAmount} symbol={call.transferTokenSymbol} decimal={call.transferTokenDecimals} />
```

it looks like you can concat those into a new component to display the amount (and you can include price lateR)

apply to all below